File: backend/src/main/java/com/gnome/crossing/backend/model/Game.java
package com.gnome.crossing.backend.model;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonPropertyOrder;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author davidjmartin
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@JsonPropertyOrder({ "id", "description", "startDate", "startTimes", "itemLocations", "players" })
public class Game {

    private long id;

    private String startDate;
    private String description;

    private List<Player> players;

    private List<ItemLocation> itemLocations;

    private List<StartTime> startTimes;

}

File: frontend/src/app/model/Game.ts
import { ItemLocation } from './ItemLocation';
import { Player } from './Player';
import { StartTime } from './StartTime';

export interface Game {
    id: number;

    startDate: string;
    description: string;    

    players: Player[];
    itemLocations: ItemLocation[];
    startTimes: StartTime[];

}